test/utils/widget_pump.dart

```diff
@@ -9,3 +11,15 @@ Future<void> pumpMaterialScoped(WidgetTester tester, Widget widget) => tester.pu
         ),
       ),
     );
+
+/// Wraps a [pumpMaterialScoped] with a [ProviderScope] that overrides [themeController].
+Future<void> pumpMaterialScopedWithTheme(WidgetTester tester, Widget widget) => tester.pumpWidget(
+      ProviderScope(
+        overrides: [themeController.overrideWithValue(ThemeController())],
+        child: MaterialApp(
+          home: Scaffold(
+            body: widget,
+          ),
+        ),
+      ),
+    );
```

Comment:
```suggestion
/// Wraps a [pumpMaterialScoped] with a [ProviderScope].
Future<void> pumpProviderScoped(WidgetTester tester, Widget widget, [ List<?> overrides ]) => tester.pumpWidget(
      ProviderScope(
        overrides: overriddes ?? [],
        child: MaterialApp(
          home: Scaffold(
            body: widget,
          ),
        ),
      ),
    );
```

I think there is no need to override the `ThemeController`, as the `themeController` already pass the exact same implemented value that you provided on the override.

Also I think allowing an arbitrary list of overrides would be best (don't know the exact type, so I've put a `?`).

Reply:
I'm not sure what you suggested about removing the `ThemeController` override, but without it it's not possible to test the widget because it uses the hooks to fetch the controller from the Widgets tree. If I misunderstood the suggestion please let me know.

Reply:
> it uses the hooks to fetch the controller from the Widgets tree

It's provided by [default by `themeController`](https://github.com/olmps/memo/blob/custom-button/lib/application/theme/theme_controller.dart#L10) (variable containing the theme provider).

Reply:
> it uses the hooks to fetch the controller from the Widgets tree

It's provided by [default by `themeController`](https://github.com/olmps/memo/blob/custom-button/lib/application/theme/theme_controller.dart#L10) (variable containing the theme provider).

Reply:
But we are not using it directly, we use it through [`useTheme`](https://github.com/olmps/memo/blob/302ead3a120f1b10a952397ebbcec4f089dd3d2e/lib/application/theme/theme_controller.dart#L90) which trie to read it from the context scope.